tests/unit/test_charm.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+
+"""Test default charm events such as upgrade charm, install, etc."""
+
+import unittest
+from unittest.mock import PropertyMock, patch
+
+import ops.testing
+from charm import SlurmdbdCharm
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+ops.testing.SIMULATE_CAN_CONNECT = True
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self) -> None:
+        self.harness = Harness(SlurmdbdCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    @unittest.expectedFailure
+    def test_install_success(self) -> None:
+        """Test that slurmdbd can successfully be installed.
+
+        Notes:
+            This method is expected to fail due to the 'version' file missing.
```

Comment:
We would previously generate the version file on [build](https://github.com/omnivector-solutions/slurm-charms/blob/master/Makefile#L43,L44) using `git describe --tags --dirty > version`.

How do you feel about adding a build command and env to the `tox.ini` that generates this file and runs `charmcraft pack`?

Reply:
I can look into this. I did have a version command added to the `tox.ini` file in the original `ov-slurm-charms` repository:

```ini
[testenv:version]
description = Create or update version file.
commands =
    bash -c "git describe --tags --always > version"
```

For the integration tests I will be packing a test charm, so I will look to include this tox environment in the next pull request.